src/patchelf.cc

```diff
@@ -1000,8 +1000,9 @@ void ElfFile<ElfFileParamNames>::normalizeNoteSegments()
             size_t size = 0;
             for (const auto & shdr : shdrs) {
                 if (rdi(shdr.sh_type) != SHT_NOTE) continue;
-                if (rdi(shdr.sh_offset) != curr_off) continue;
+                if (rdi(shdr.sh_offset) != roundUp(curr_off, rdi(shdr.sh_addralign))) continue;
```

Comment:
This looks correct to me. For reference this is the calculation that gold performs: https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=bfd/elf.c;h=851440b765284763b373739444073f732d31f60b;hb=HEAD#l4244